### ISSUE TITLE
Intermittent errors with yarn dev yielding "connect ECONNREFUSED 127.0.0.1:62919"

### ISSUE THREAD

**Issue by ddeisadze** [wxt-dev/wxt#460]
### Describe the bug

When I am developing, sometimes I see the "wxt" command exists with this message (does not really say why):

```
[2:22:20 PM] WXT 0.16.6
[2:22:20 PM] ✔ Started dev server @ http://localhost:3000
[2:22:20 PM] ℹ Pre-rendering chrome-mv3 for development with Vite 5.1.2
node_modules/jotai/esm/react.mjs (1:0) Error when using sourcemap for reporting an error: Can't resolve original location of error.
⠙ [4/5] logout2:22:25 PM [vite] Pre-transform error: Failed to load url /popup.js (resolved id: /popup.js). Does the file exist?
[2:22:25 PM] ✔ Built extension in 5.293 s
  ├─ .output/chrome-mv3/manifest.json               1.06 kB  
  ├─ .output/chrome-mv3/popup.html                  607 B    
  ├─ .output/chrome-mv3/background.js               7.55 MB  
  ├─ .output/chrome-mv3/content-scripts/jobDesc.js  12.41 MB 
  ├─ .output/chrome-mv3/content-scripts/login.js    175.73 kB
  ├─ .output/chrome-mv3/content-scripts/logout.js   177.28 kB
  ├─ .output/chrome-mv3/popup.js                    7.39 kB  
  ├─ .output/chrome-mv3/icon/128.png                9.04 kB  
  ├─ .output/chrome-mv3/icon/16.png                 2.71 kB  
  ├─ .output/chrome-mv3/icon/256.png                24.33 kB 
  ├─ .output/chrome-mv3/icon/48.png                 2.96 kB  
  └─ .output/chrome-mv3/wxt.svg                     1.07 kB  
Σ Total size: 20.37 MB                                               
[2:22:51 PM] ✖ Command failed after 30.7 s

[2:22:51 PM]  ERROR  connect ECONNREFUSED 127.0.0.1:62919

  at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1595:16) 
```

### To Reproduce

Reproduction might be difficult to attach, I have multiple shared libraries. Let me know if i should add it

**Comment by aklinker1** [comment]
I understand reproduction might be difficult, that's fine. As others encounter this issue, they can add more context. But please add environment information (and I'll re-open). I need to know which OS you're using.

```
npx envinfo --system --browsers --binaries --npmPackages wxt,vite
```

That said, this is likely an upstream issue with `web-ext`.

**Comment by micksabox** [comment]
Same issue for me. 25seconds after startup I get this error:

```
✖ Command failed after 25.8 s                                                    6:49:36 p.m.

 ERROR  connect ECONNREFUSED 127.0.0.1:49917                                      6:49:36 p.m.

    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1595:16)
```

```
  System:
    OS: macOS 15.3.2
    CPU: (10) arm64 Apple M2 Pro
    Memory: 147.38 MB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.11.1 - ~/.nvm/versions/node/v20.11.1/bin/node
    npm: 10.2.4 - ~/.nvm/versions/node/v20.11.1/bin/npm
    pnpm: 9.15.2 - /opt/homebrew/bin/pnpm
    bun: 1.0.22 - ~/.bun/bin/bun
    Watchman: 2024.12.02.00 - /opt/homebrew/bin/watchman
  Browsers:
    Brave Browser: 134.1.76.80
    Chrome: 134.0.6998.118
    Safari: 18.3.1
  npmPackages:
    wxt: ^0.19.29 => 0.19.29
```

**Comment by eddiew** [comment]
+1

**Comment by sdz123** [comment]
same problem

**Comment by aklinker1** [comment]
I've been experimenting with an alternative to `web-ext`, and I think the problem is that chrome can't open multiple instances from the same directory at the same time. When spawning the second process, it sees there's another instance of chrome open and closes immediately. Then, `web-ext` waits and is never able to connect to the debugger port of the second instance.

So try closing any other open chrome windows. I like to use Chrome Canary during development so it doesn't interfere with my normal chrome instance. That makes it easy to track if a window is already open.

**Comment by sdz123** [comment]
> [I've been experimenting with an alternative to `web-ext`](https://github.com/wxt-dev/wxt/pull/1566), and I think the problem is that chrome can't open multiple instances from the same `--user-data-dir` at the same time. When spawning the second process, it sees there's another instance of chrome open and closes immediately. Then, `web-ext` waits and is never able to connect to the debugger port of the second instance.
> 
> So try closing any other open chrome windows. I like to use Chrome Canary during development so it doesn't interfere with my normal chrome instance. That makes it easy to track if a window is already open.

i removed this config in wxt.config.js,it works for me:

 chromiumArgs: ['--user-data-dir=./.wxt/browser-data'],

**Comment by rpayanm** [comment]
> So try closing any other open chrome windows.

Thank you! That worked. I did not have any Chrome windows open (visually), but there was some process running in the background. On my Mac, I opened Activity Monitor and found these processes:

<img width="366" height="193" alt="Image" src="https://github.com/user-attachments/assets/995abe29-fb51-45e4-9abd-508e9f0b75d5" />

I closed these two "Google Chrome" processes, and I was able to run `npm run dev` again without any issues.
